presto-main/src/main/java/io/prestosql/sql/planner/SubqueryPlanner.java

```diff
@@ -226,7 +226,8 @@ private PlanBuilder appendScalarSubqueryApplyNode(PlanBuilder subPlan, SubqueryE
             subPlan.getTranslations().put(coercion, coercionSymbol);
         }
 
-        return appendLateralJoin(subPlan, subqueryPlan, scalarSubquery.getQuery(), correlationAllowed, LateralJoinNode.Type.LEFT);
+        // The subquery's EnforceSingleRowNode always produces a row, so the join is effectively INNER
```

Comment:
Maybe we could have an optimizer rule which might handle cases like this.

Reply:
Yes, I think we could have such a rule especially for JoinNode.

Reply:
Good idea. We can do that as a separate change later.